### ISSUE TITLE
stopped working again

### ISSUE THREAD

**Issue by pcause** [pstadler/ticker.sh#45]
Think it is the cookie stuff.  It stopped on one machine a few days ago and I copied the cook cache from another machine and it started working.  Now fails on all systems

**Comment by appatalks** [comment]
This does appear to be [wider spread](https://stackoverflow.com/questions/76059562/yahoo-finance-api-get-quotes-returns-invalid-cookie) than just to this piece of code. In that [same thread](https://stackoverflow.com/a/76356578) a user reports a different endpoint as still valid, but requires some adjustment to account for the change in results. 

Not sure how long that will last. Seems like Yahoo might be playing around with their APIs.

This image appears when navigating the URL reported in the [API_ENDPOINT](https://query1.finance.yahoo.com/v7/finance/quote?lang=en-US&region=US&corsDomain=finance.yahoo.com) 
> {"finance":{"result":null,"error":{"code":"Unauthorized","description":"Invalid Crumb. For Developers - https://bit.ly/yahoo-finance-api-feedback"}}}

![](https://github.com/pstadler/ticker.sh/assets/4163156/ce59e3b7-dfe2-4dbd-b783-4ac5b5e36c15)

Followed by a survey ending with:
![](https://github.com/pstadler/ticker.sh/assets/4163156/259ad417-35dc-4602-88ed-53f4ef82195b)

:disappointed:

**Comment by pstadler** [comment]
Here we go. Everything that happened during the last couple of weeks pointed towards this.

**Comment by phish0r** [comment]
Yep.  Mine stopped working again as well.  Is there a chance we can switch to Google Finance or is that not an option

**Comment by pcause** [comment]
there is tickr and from a quick look at code it uses yahoo.  it is working.  not sure if the code can provide a hint as to what to change to get working again

**Comment by phish0r** [comment]
so I guess we're just waiting at this point to see what yahoo does?  there does appear to be some endpoints that are still active, like: https://query1.finance.yahoo.com/v8/finance/chart/AAPL for example.  /quote seems to be dead though.

**Comment by pstadler** [comment]
This pretty much sums it up, unfortunately. I‘m looking for alternatives, but haven’t been able to find any so far.

**Comment by appatalks** [comment]
I created [PR#46](https://github.com/pstadler/ticker.sh/pull/46) if y'all would like to give that a play with :penguin:

**Comment by pcause** [comment]
> I created [PR#46](https://github.com/pstadler/ticker.sh/pull/46) if y'all would like to give that a play with penguin

seems to be working for me

**Comment by phish0r** [comment]
I

> > I created [PR#46](https://github.com/pstadler/ticker.sh/pull/46) if y'all would like to give that a play with penguin
> 
> seems to be working for me

I can confirm it's working for me as well.

**Comment by g4570n** [comment]
If it is working, will the  PR#46 merge be done?

**Comment by pstadler** [comment]
I‘m honestly not convinced that a request per ticker is a good solution, even if it‘s the only way to make it work. Thoughts?

**Comment by realpshev** [comment]
as you say, the only way to make it work.  i can live with this versus not having the capability.

**Comment by pcause** [comment]
PR #46 needs to add support for NOCOLOR

**Comment by appatalks** [comment]
> PR #46 needs to add support for NOCOLOR

I see in the original code:

```bash
if [ -z "$NO_COLOR" ]; then
  : "${COLOR_BOLD:=\e[1;37m}"
  : "${COLOR_GREEN:=\e[32m}"
  : "${COLOR_RED:=\e[31m}"
  : "${COLOR_RESET:=\e[00m}"
fi
```

I'll try to find some time to learn how to get that in place :)

**Comment by pcause** [comment]
in my use case it looks like it works.  thanks

**Comment by pstadler** [comment]
it feels like every solution using Yahoo Finance will be short-lived from now on.

**Comment by appatalks** [comment]
> the pr 46 version seems to have stopped working today

I wonder if it might be location based? At the moment, it is still working for me. 
What error are you getting? and I'll see if I can recreate/play with it on my end too :)

**Comment by pcause** [comment]
interesting.  it worked on my linux system but not wsl2 on windows

**Comment by appatalks** [comment]
@pcause sorry I haven't responded in a bit for this, for WSL issue, I am afraid I am not have much success getting my windows & hardware to work with wsl in general to effectively troubleshoot (this is a me issue on my part) :(

**Comment by pcause** [comment]
np.  figured it out.  was the python issue that someone else posted about separately

**Comment by appatalks** [comment]
:wave: Team,

I adjusted the script to not use ```python```, and use ```bc``` instead.

At first I was going to try to stick with python, but then I was worried about users who require ```python2``` and then I was trying to find logic to get around that, tl'dr ... it got messy.

**Comment by pcause** [comment]
pr45 version - there is a small bug.  then computing the percentage change it should be scale=4.  bc will calculate to 4 decimal places and the multiply result will give you a percent and factional percent result with this change.

Is there some way to get a clean version.  it is hard to find the latest.

**Comment by appatalks** [comment]
👋 @pcause 

I have [my fork here](https://github.com/appatalks/ticker.sh/blob/master/ticker.sh) with the ```master``` branch currently what I am running for my day-to-day. (In all openness, I am uncertain if it's bug free, but just let me know and Ill be happy to review)

**Comment by pcause** [comment]
the one I have uses bc but that version doesn't.  it has:

percentChange=$(printf "%.2f" $(echo "scale=4; (($currentPrice - $previousClose) / $previousClose) * 100" | bc))

seems you changed at some point?

**Comment by appatalks** [comment]
Ah yes! I remember now, yes, I was also using ```bc``` originally, but I kept running against math'ing errors that I couldn't quite figure out. Ran it through AI and it suggested my issue was something to do with ```bc and floating-point arithmetic and routing```  or some other way above my head. The AI made the ```awk``` suggestion to use instead and so far 🤞 🤞

**Comment by pcause** [comment]
interesting.  i've been using the bc version without issues other than the one above.

**Comment by appatalks** [comment]
I'd be happy to revert and try that ```scale=2``` variable. Not sure if that was something I tried to adjust when trying to get it to work.  Might need a bit of time to play with it.

**Comment by pcause** [comment]
not worth the effort.  i'll grab your version but stash away the one i'm using

**Comment by pstadler** [comment]
Here an update you all deserve to hear.
I'm not happy with the available workarounds. The only available endpoint makes ticker.sh painfully slow (see #46).
Until another provider comes around which gives some endpoints and data for free, I'm considering this project halted.

**Comment by lm2lm2** [comment]
hello
is it possible to make it work back? :)

**Comment by ghost** [comment]
I appreciate your effort on this, but could you please put a nice, big, juicy banner on the landing page that states, "I'm considering this project halted?"... just so we don't waste people's time.  Thanks!

**Comment by pstadler** [comment]
@adanhawth Makes sense. Added.

**Comment by lm2lm2** [comment]
that's sad ; no way or other working project to have a similar tool?
thank you

**Comment by andymiller-og** [comment]
I still come here to see if there's any sort of progress or alternative option. I guess not. :-(

**Comment by appatalks** [comment]
@andymiller-og 👋 
Give this one a try: [appatalks/ticker.sh](https://github.com/appatalks/ticker.sh)
